pkg/certifier/components/root_package/root_package.go

```diff
@@ -88,12 +88,17 @@ func getCompHelper(ctx context.Context, client graphdb.Client, parentPurl string
 		if !ok {
 			return nil, errors.New("failed to cast to node type")
 		}
+		if visited[foundDep.Id] {
+			continue
+		}
+		visited[foundDep.Id] = true
```

Comment:
There is a very small optimization where the root is not bootstrapped in the visited map, this can be moved to the top of the function (with the parentPurl) in order to remediate that, else lgtm

Reply:
@lumjjb A friendly ping

Reply:
Thanks, done!